src/tribol/geom/GeomUtilities.cpp

```diff
@@ -1286,20 +1301,23 @@ FaceGeomError CheckPolySegs( const real* const RESTRICT x, const real* const RES
 } // end CheckPolySegs()
 
 //------------------------------------------------------------------------------
-void PolyReorder( real* const RESTRICT x, real* const RESTRICT y, const int numPoints )
+TRIBOL_HOST_DEVICE bool PolyReorder( RealT* const x, RealT* const y, const int numPoints )
 {
 
    if (numPoints<3)
    {
+#ifdef TRIBOL_USE_HOST
       SLIC_DEBUG("PolyReorder: numPoints (" << numPoints << ") < 3.");
-      return;
+#endif
+      return false;
    }
 
-   real xC, yC, zC;
-   real * z = nullptr;
-   real proj [numPoints - 2];
+   RealT xC, yC, zC;
+   RealT * z = nullptr;
+   constexpr int max_nodes_per_overlap = 8;
+   RealT proj [max_nodes_per_overlap - 2];
```

Comment:
Same comment about whether we want to initialize values in arrays sized with some max length?